drivers/adc/ad463x/ad463x.h

```diff
@@ -206,9 +206,9 @@ struct ad463x_init_param {
 	struct no_os_gpio_init_param *gpio_cnv;
 	struct no_os_gpio_init_param *gpio_pgia_a0;
 	struct no_os_gpio_init_param *gpio_pgia_a1;
-#ifndef USE_STANDARD_SPI
 	/** PWM */
 	struct no_os_pwm_init_param *trigger_pwm_init;
```

Comment:
This sounds restricting. What if the platform doesn't have PWM?
I see that CNV (often triggered by a PWM) and CS must be synchronized for data read in CS zone 2.
But maybe better to not assume all steups will want to use a PWM to trigger CNV. It looks like one could use a GPIO to pulse CNV and CS would not need be that much synchronized with CNV (as long as the transfer is done after BUSY goes low (zone 1 capture)).
I'd check the init params to decide what to use for CNV instead of assuming a PWM is there.

Reply:
What you mention is actually how it is working:

 the pwm is needed in two cases:
- spi engine offload is enabled
- using standard spi with dma to toggle the CNV without cpu intervention.

otherwise in the "standard spi" mode without dma a gpio is used to toggle
cnv and antother gpio to toggle cs. 

the driver checks for these two conditions on init to enable the pwm:

see: https://github.com/analogdevicesinc/no-OS/pull/2225/files/06d1222f7f7d125869f526cd69e074ed5a1a5a10..cf315b0fa0fd565db9b3c7c4de314b49b2079a43#diff-f169717926e4e3f8e673fc4f56466d92742b7e39635579b62d80259fc2e8de60R848
```
        if (dev->spi_dma_enable || dev->offload_enable) {
                ret = no_os_pwm_init(&dev->trigger_pwm_desc,
                                     init_param->trigger_pwm_init);
                if (ret != 0)
                        goto error_spi;
        }

```

the ad463x_read_data function will:
 - use offload when USE_STANDARD_SPI is not defined.
 - use "ad463x_read_data_dma" when spi_dma_enable is set
 - use "ad463x_read_single_sample" when spi_dma_enable is not set


```
static int32_t ad463x_read_single_sample(struct ad463x_dev *dev,
                uint32_t *ch0_out,  uint32_t *ch1_out)
{
        uint8_t data[8] = {0};
        int ret;

        if (!dev)
                return -EINVAL;

        ret = no_os_gpio_set_value(dev->gpio_cnv, NO_OS_GPIO_HIGH);
        if (ret)
                return ret;

        ret = no_os_gpio_set_value(dev->gpio_cnv, NO_OS_GPIO_LOW);
        if (ret)
                return ret;

        ret = no_os_spi_write_and_read(dev->spi_desc, data, dev->read_bytes_no);
        if (ret)
                return ret;

        ad463x_pext_sample(dev, data, dev->read_bytes_no, ch0_out, ch1_out);

        return 0;
}

```

Reply:
Oh, I see now. Okay, that looks good.
I think I got misled by the commit message.

Instead of
> Enable the CNV pwm for standard spi and  Use the dma sync function
> to do sample reads.

I would say something more along the lines of:

Enable DMA transfers with standard SPI when the SPI controller is DMA capable. In that case, request and use a PWM to trigger conversions on the CNV pin.

Reply:
ok.. ill re-word it.